src/callbacks.cc

```diff
@@ -458,12 +458,13 @@ void async_cb_handler(uv_async_t *handle)
 {
   NotifInfo *notif;
 
-  mutex::scoped_lock sl(zqueue_mutex);
-
-  while (!zqueue.empty()) {
-    notif = zqueue.front();
-    zqueue.pop();
-    zqueue_mutex.unlock();
+  while (true) {
```

Comment:
this while loop looks wrong: it consumes all the items from the queue and simply breaks out then it has popped the last item.

Reply:
I'm not sure how you came to that conclusion. For each item in the queue, it:
- pops it off (the only operation along with the empty check under the mutex)
- calls `handleNotification` or `handleControllerCommand`
- deletes the notification